packages/core-api/src/apis/definitions/auth.ts

```diff
@@ -143,6 +143,30 @@ export type OpenIdConnectApi = {
   logout(): Promise<void>;
 };
 
+export type ProfileInfoOptions = {
+  /**
+   * If this is set to true, the user will not be prompted to log in,
+   * and an empty profile will be returned if there is no existing session.
+   *
+   * This can be used to perform a check whether the user is logged in, or if you don't
+   * want to force a user to be logged in, but provide functionality if they already are.
+   *
+   * @default false
+   */
+  optional?: boolean;
+};
+
+export type ProfileInfoApi = {
+  getProfile(options?: ProfileInfoOptions): Promise<ProfileInfo | undefined>;
+};
+
+export type ProfileInfo = {
+  provider: string;
```

Comment:
I'd say we can remove provider from the info, I don't think you're using it either? With the current API you'll always know which provider you're dealing with when it's being consumed, and if you want to treat the in a more generic way at some point you could add in the information later, e.g.

```ts
const allTheProviders: {api: ProfileInfoApi, provider: string}[] = [
  {api: googleAuthApi, provider: 'google'},
  {api: githubAuthApi, provider: 'github'},
]
```